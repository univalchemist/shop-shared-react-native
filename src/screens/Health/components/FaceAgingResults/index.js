import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Slider } from 'react-native-elements';
import {
  Box,
  Flex,
  Image,
  Loader,
  ScreenHeadingText,
  Text,
} from '@wrappers/components';
import theme from '@theme';
import PropTypes from 'prop-types';
import { useIntl } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard.js';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { categories, logAction } from '@store/analytics/trackingActions';

const styles = StyleSheet.create({
  container: { paddingHorizontal: 32, paddingBottom: 32 },
});

export const FaceAgingResults = ({ faceAging }) => {
  const intl = useIntl();
  const [selectedAgeIndex, setSelectedAgeIndex] = useState(0);
  return (
    <View style={styles.container}>
      {faceAging.isError ? (
        renderFaceAgingErrorPanel({ intl })
      ) : faceAging.faceAgingIsDone &&
        faceAging.expectedTotalResults === faceAging.currentTotalResults ? (
        renderFaceAgingResults(
          faceAging,
          selectedAgeIndex,
          setSelectedAgeIndex,
          intl,
        )
      ) : (
        <>
          <ScreenHeadingText mb={2}>
            {intl.formatMessage({ id: 'futureMeTitle' })}
          </ScreenHeadingText>
          <CardContainer
            theme={theme}
            justifyContent="center"
            color={theme.colors.white}
            flexGrow={1}
          >
            <Loader
              primary
              textAlign={'center'}
              loadingText={intl.formatMessage({
                id: 'health.healthScoreHistory.lifestyleImage.processing',
                defaultMessage:
                  'Analysing in progress. This may take a little while. Please do not navigate away.',
              })}
            />
          </CardContainer>
        </>
      )}
    </View>
  );
};

const renderFaceAgingResults = (
  faceAging,
  selectedAgeIndex,
  setSelectedAgeIndex,
  intl,
) => {
  let faceAges = [];
  if (faceAges.length === 0) {
    faceAges = Object.keys(faceAging.results).sort();
  }
  const selectedAge = faceAges[selectedAgeIndex];
  const selectedAgeForDisplay = selectedAge == '67' ? '65' : selectedAge;

  return faceAges.length > 0 ? (
    <>
      <ScreenHeadingText mb={2}>
        {intl.formatMessage({ id: 'futureMeTitle' })}
      </ScreenHeadingText>
      <Box>
        <Text
          accessible={true}
          accessibilityLabel={intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
          color={theme.colors.gray[0]}
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          mt={8}
          mb={8}
        >
          {intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
        </Text>
        {renderFaceAgingImages(selectedAge, faceAging, intl)}
        <Box mt={16}>
          <Slider
            style={{ width: '100%', height: 40 }}
            step={1}
            minimumValue={0}
            maximumValue={faceAges.length - 1}
            thumbTintColor={theme.colors.black}
            minimumTrackTintColor={theme.colors.black}
            maximumTrackTintColor={theme.colors.gray[4]}
            onValueChange={faceAgeIndex => {
              setSelectedAgeIndex(faceAgeIndex);
            }}
            onSlidingComplete={() => {
              logAction({
                category: categories.LIFESTYLE_OVERVIEW,
                action: `Slider age ${selectedAgeForDisplay}`,
              });
            }}
          />
          <Flex flex={1} flexDirection="row" />
        </Box>
        <Text
          accessibilityLabel={intl.formatMessage({
            id: 'health.faceAgingDescription',
          })}
          accessible={true}
          fontSize={14}
          mt={8}
          mb={16}
        >
          {intl.formatMessage({ id: 'health.faceAgingDescription' })}
        </Text>
      </Box>
    </>
  ) : (
    <Box />
  );
};

const renderFaceAgingImages = (selectedAge, faceAging, intl) => {
  return (
    <>
      {Object.keys(faceAging.results).map((faceAge, index) => {
        return (
          <Flex
            key={index}
            flexDirection="row"
            style={faceAge === selectedAge ? {} : { display: 'none' }}
          >
            <Box flex={1} marginRight={8}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingHealthyLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].healthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingHealthyLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingHealthyLifestyle' })}
                </Text>
              </Box>
            </Box>
            <Box flex={1}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingCurrentLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].unhealthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingCurrentLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingCurrentLifestyle' })}
                </Text>
              </Box>
            </Box>
          </Flex>
        );
      })}
    </>
  );
};

const renderFaceAgingErrorPanel = ({ intl }) => (
  <Box>
    <ScreenHeadingText mb={2}>
      {intl.formatMessage({ id: 'futureMeTitle' })}
    </ScreenHeadingText>
    <ErrorCard color={theme.colors.white} />
  </Box>
);

const mapStateToProps = ({ health: { faceAging } }) => ({
  faceAging,
});

const enhance = compose(connect(mapStateToProps));

export default enhance(FaceAgingResults);

FaceAgingResults.propTypes = {
  faceAging: PropTypes.shape({
    faceAgingIsDone: PropTypes.bool,
    expectedTotalResults: PropTypes.number,
    currentTotalResults: PropTypes.number,
    results: PropTypes.shape({}),
  }),
};
