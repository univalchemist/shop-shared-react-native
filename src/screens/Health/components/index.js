/* istanbul ignore file */

export {
  CallToActionLink,
  CardContainer,
  CardLoader,
  ErrorCard,
  TipCard,
} from './widgets';

export { default as ProductRecommendation } from './ProductRecommendation';
export { default as LifestyleTipsModal } from './LifestyleTipsModal';
export { default as LifestyleResults } from './LifestyleResults';
export { default as HealthScoreHistory } from './HealthScoreHistory';
export { default as HealthScore } from './HealthScore';
export { default as HealthResultCard } from './HealthResultCard';
export { default as GeneralTips } from './GeneralTips';
export { default as FaceAgingResults } from './FaceAgingResults';
