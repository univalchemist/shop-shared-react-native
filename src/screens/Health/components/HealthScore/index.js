import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import HealthScoreError from './HealthScoreError';
import HealthScoreDisplay, { HealthScoreLoading } from './HealthScoreDisplay';
import { PromiseStatus } from '@middlewares';
import { Card } from '@wrappers/components';
import { getUserHealthScore } from '@store/health/actions';

export const HealthScore = ({ healthScore = {}, getUserHealthScore }) => {
  useEffect(() => {
    const fetchUserHealthScore = async () => {
      await getUserHealthScore();
    };
    fetchUserHealthScore();
  }, [getUserHealthScore]);

  const { score, status } = healthScore;
  const isValidScore = !isNaN(parseInt(score, 10));
  const showHealthScore = () => {
    if (status === PromiseStatus.START) {
      return <HealthScoreLoading />;
    }
    if (status === PromiseStatus.ERROR || !isValidScore) {
      return <HealthScoreError />;
    }
    if (status === PromiseStatus.SUCCESS) {
      return <HealthScoreDisplay score={score} />;
    }
    return null;
  };

  return (
    <Card bg="white" flexGrow={1} px={4} pt={4} pb={24}>
      {showHealthScore()}
    </Card>
  );
};

const mapStateToProps = ({ health: { healthScore } }) => ({
  healthScore,
});

export default connect(mapStateToProps, { getUserHealthScore })(HealthScore);
