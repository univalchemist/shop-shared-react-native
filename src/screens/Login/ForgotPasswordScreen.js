import {
  Box,
  Button,
  Container,
  Loader,
  ScrollView,
  Text,
  ScreenHeadingText,
} from '@wrappers/components';
import { InputField } from '@wrappers/components/FinalForm';
import { useIntl, useTheme } from '@wrappers/core/hooks';
import { forgotPassword } from '@store/user/actions';
import PropTypes from 'prop-types';
import React, { useRef } from 'react';
import { SafeAreaView } from 'react-native';
import { connect } from 'react-redux';
import { Form } from 'react-final-form';
import { debounceAlert } from '@utils';
import {
  validateCompanyCharacters,
  validateEmail,
  validateRequiredCompanyName,
} from '@wrappers/core/validations';
import { getLocalizeServerError } from '@utils/localizeServerError';

const ForgotPasswordScreen = ({ forgotPassword, route }) => {
  const theme = useTheme();
  const intl = useIntl();
  const params = route?.params;

  const initialValues = {
    clientId: params?.clientId,
    username: params?.username,
  };

  const clientnameRef = useRef(null);
  const usernameRef = useRef(null);

  const onSubmit = async ({ clientId, username }) => {
    try {
      await forgotPassword({ clientId, username });

      debounceAlert({
        subject: intl.formatMessage({ id: 'forgotPassword.successTitle' }),
        message: intl.formatMessage(
          { id: 'forgotPassword.successText' },
          { email: username },
        ),
      });
    } catch (error) {
      const { subject, message } = getLocalizeServerError(
        error,
        {
          subjectPrefix: 'serverErrors.forgot.subject',
          prefix: 'serverErrors.forgot',
        },
        intl,
      );
      debounceAlert({
        subject,
        message,
      });
    }
  };

  return (
    initialValues && (
      <Form
        initialValues={initialValues}
        onSubmit={onSubmit}
        render={({ handleSubmit, submitting }) =>
          submitting ? (
            <Loader />
          ) : (
            <Box
              as={SafeAreaView}
              height="100%"
              backgroundColor={theme.backgroundColor.default}
            >
              <ScrollView>
                <Container mt={4}>
                  <Box pb={2}>
                    <ScreenHeadingText>
                      {intl.formatMessage({
                        id: 'forgotPassword.introText',
                      })}
                    </ScreenHeadingText>
                  </Box>
                  <Box>
                    <Text>
                      {intl.formatMessage({
                        id: 'forgotPassword.instructions',
                      })}
                    </Text>
                  </Box>

                  <Box mt={4}>
                    <Box>
                      <InputField
                        ref={clientnameRef}
                        validate={[
                          validateRequiredCompanyName,
                          validateCompanyCharacters,
                        ]}
                        onSubmitEditing={() => usernameRef.current.focus()}
                        name="clientId"
                        autoCapitalize="none"
                        hint={intl.formatMessage({ id: 'loginClientNameHint' })}
                        label={intl.formatMessage({ id: 'loginClientIdLabel' })}
                        returnKeyType="next"
                        testID="clientId"
                        customStyles={theme.customInputStyles}
                      />
                    </Box>
                    <Box>
                      <InputField
                        validate={validateEmail}
                        ref={usernameRef}
                        name="username"
                        autoCapitalize="none"
                        keyboardType="email-address"
                        label={intl.formatMessage({ id: 'loginUsernameLabel' })}
                        returnKeyType="next"
                        testID="username"
                        customStyles={theme.customInputStyles}
                      />
                    </Box>
                    <Box>
                      <Button
                        disabled={submitting}
                        primary
                        onPress={handleSubmit}
                        title={intl.formatMessage({
                          id: 'forgotPassword.submitButtonLabel',
                        })}
                      />
                    </Box>
                  </Box>
                </Container>
              </ScrollView>
            </Box>
          )
        }
      />
    )
  );
};

ForgotPasswordScreen.propTypes = {
  forgotPassword: PropTypes.func.isRequired,
};

export default connect(null, { forgotPassword })(ForgotPasswordScreen);
