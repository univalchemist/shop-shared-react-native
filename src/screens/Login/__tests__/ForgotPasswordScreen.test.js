import { Button, Loader } from '@wrappers/components';
import { render } from '@testUtils';
import React from 'react';
import { TextInput } from 'react-native';
import {
  act,
  fireEvent,
  flushMicrotasksQueue,
} from 'react-native-testing-library';
import messages from '@messages/en-HK.json';
import ForgotPasswordScreen from '../ForgotPasswordScreen';
import { debounceAlert } from '@utils';

jest.mock('@utils', () => ({
  ...require.requireActual('@utils'),
  debounceAlert: jest.fn(),
}));

jest.useFakeTimers();

const route = {
  params: {
    clientId: 'clientIdFromParam',
    username: 'b@c.com',
  },
};

const api = {
  forgotPassword: jest.fn(({ username }) => {
    if (username === 'error@b.com') {
      return Promise.reject();
    }
    return Promise.resolve({ data: {} });
  }),
};

const validValues = {
  clientId: 'test',
  username: 'a@b.com',
};

const valuesWithErrorKey = {
  clientId: 'test',
  username: 'error@b.com', // triggers error
};

let screen, fields;

describe('ForgotPasswordScreen', () => {
  beforeEach(async () => {
    jest.clearAllMocks();
    jest.clearAllTimers();
    [screen] = render(<ForgotPasswordScreen route={route} />, {
      api,
    });
    await flushMicrotasksQueue();
    fields = screen.queryAllByType(TextInput);
  });

  it('should render display text, all field inputs and button', async () => {
    expect(
      screen.getByText(messages['forgotPassword.introText']),
    ).toBeDefined();
    expect(
      screen.getByText(messages['forgotPassword.instructions']),
    ).toBeDefined();
    expect(fields).toHaveLength(2);
    expect(
      screen.getByProps({
        title: messages['forgotPassword.submitButtonLabel'],
      }),
    ).toBeDefined();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should validate company id and email', async () => {
    act(() => {
      fields.forEach(x => {
        fireEvent.changeText(x, '');
        fireEvent(x, 'blur');
      });
    });

    expect(screen.getByText(messages.invalidCompanyName)).toBeTruthy();
    expect(screen.getByText(messages.invalidEmail)).toBeTruthy();

    act(() => {
      fireEvent.changeText(fields[0], 'companynamewithinvalidchar.');
    });
    expect(screen.getByText(messages.invalidCompanyCharacters)).toBeTruthy();
    const submit = screen.getByType(Button);
    act(() => {
      fireEvent.press(submit);
    });
    expect(api.forgotPassword).not.toHaveBeenCalled();
  });

  it('should call forgotPassword api on submit', async () => {
    const submitButton = screen.getByProps({
      title: messages['forgotPassword.submitButtonLabel'],
    });

    act(() => {
      fields.forEach(field => {
        const name = field.props.name;
        fireEvent.changeText(field, validValues[name]);
      });
      fireEvent.press(submitButton);
    });

    await flushMicrotasksQueue();

    expect(api.forgotPassword).toHaveBeenCalledWith(validValues);
  });

  it('should display a loader while api request is being made', async () => {
    const submitButton = screen.getByProps({
      title: messages['forgotPassword.submitButtonLabel'],
    });
    act(() => {
      fields.forEach(field => {
        const name = field.props.name;
        fireEvent.changeText(field, validValues[name]);
      });
      fireEvent.press(submitButton);
    });
    expect(screen.queryByType(Loader)).toBeDefined();
    await flushMicrotasksQueue();
    expect(screen.queryByType(Loader)).toBeNull();
  });

  it('should invoke success alert on successful api response', async () => {
    const submitButton = screen.getByProps({
      title: messages['forgotPassword.submitButtonLabel'],
    });
    act(() => {
      fields.forEach(x => {
        const name = x.props.name;
        fireEvent.changeText(x, validValues[name]);
      });
      fireEvent.press(submitButton);
    });

    await flushMicrotasksQueue();

    const successText = messages['forgotPassword.successText'].replace(
      '{email}',
      validValues.username,
    );
    expect(debounceAlert).toHaveBeenCalledWith({
      subject: messages['forgotPassword.successTitle'],
      message: successText,
    });
  });

  it('should invoke error alert if the API request fails', async () => {
    act(() => {
      fields.forEach(x => {
        const name = x.props.name;
        fireEvent.changeText(x, valuesWithErrorKey[name]);
      });
    });

    const submitButton = screen.getByProps({
      title: messages['forgotPassword.submitButtonLabel'],
    });
    act(() => {
      fireEvent.press(submitButton);
    });
    await flushMicrotasksQueue();

    expect(debounceAlert).toHaveBeenCalledWith({
      subject: messages['errorPanel.title'],
      message: messages['errorPanel.message'],
    });
  });
});
