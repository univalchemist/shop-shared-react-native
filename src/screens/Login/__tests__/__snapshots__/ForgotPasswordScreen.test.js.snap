// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordScreen should render display text, all field inputs and button 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor="#f9f9f9"
                        emulateUnlessSupported={true}
                        height="100%"
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#f9f9f9",
                              "height": "100%",
                            },
                          ]
                        }
                      >
                        <View
                          flex={1}
                          onLayout={[Function]}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#f9f9f9",
                                "flexBasis": 0,
                                "flexGrow": 1,
                                "flexShrink": 1,
                              },
                            ]
                          }
                        >
                          <RCTScrollView
                            automaticallyAdjustContentInsets={false}
                            contentInset={
                              Object {
                                "bottom": 0,
                              }
                            }
                            enableAutomaticScroll={true}
                            enableOnAndroid={false}
                            enableResetScrollToCoords={true}
                            extraHeight={75}
                            extraScrollHeight={0}
                            getScrollResponder={[Function]}
                            handleOnScroll={[Function]}
                            innerRef={[Function]}
                            keyboardDismissMode="on-drag"
                            keyboardOpeningTime={250}
                            keyboardShouldPersistTaps="handled"
                            keyboardSpace={0}
                            onScroll={[Function]}
                            onScrollBeginDrag={null}
                            resetKeyboardSpace={[Function]}
                            scrollEventThrottle={16}
                            scrollForExtraHeightOnAndroid={[Function]}
                            scrollIndicatorInsets={
                              Object {
                                "right": 1,
                              }
                            }
                            scrollIntoView={[Function]}
                            scrollToEnd={[Function]}
                            scrollToFocusedInput={[Function]}
                            scrollToPosition={[Function]}
                            showsVerticalScrollIndicator={true}
                            update={[Function]}
                            viewIsInsideTabBar={false}
                          >
                            <View>
                              <View
                                mt={4}
                                style={
                                  Array [
                                    Object {
                                      "marginTop": 32,
                                      "paddingBottom": 32,
                                      "paddingLeft": 32,
                                      "paddingRight": 32,
                                      "paddingTop": 32,
                                    },
                                  ]
                                }
                              >
                                <View
                                  pb={2}
                                  style={
                                    Array [
                                      Object {
                                        "paddingBottom": 8,
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.0"
                                    fontFamily="default"
                                    fontSize={26}
                                    fontWeight="light"
                                    letterSpacing={-1.5}
                                    lineHeight={37}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#212121",
                                          "fontFamily": "Arial",
                                          "fontSize": 26,
                                          "fontWeight": "300",
                                          "letterSpacing": -1.5,
                                          "lineHeight": 37,
                                        },
                                      ]
                                    }
                                  >
                                    Forgot your password?
                                  </Text>
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {},
                                    ]
                                  }
                                >
                                  <Text
                                    color="gray.1"
                                    fontFamily="default"
                                    fontSize={16}
                                    lineHeight={22}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#646464",
                                          "fontFamily": "Arial",
                                          "fontSize": 16,
                                          "lineHeight": 22,
                                        },
                                      ]
                                    }
                                  >
                                    We will email you a link to reset your password. Provide your company ID and email address.
                                  </Text>
                                </View>
                                <View
                                  mt={4}
                                  style={
                                    Array [
                                      Object {
                                        "marginTop": 32,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {},
                                      ]
                                    }
                                  >
                                    <View
                                      height={100}
                                      style={
                                        Array [
                                          Object {
                                            "height": 100,
                                          },
                                        ]
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "paddingHorizontal": 10,
                                            "paddingLeft": 0,
                                            "paddingRight": 0,
                                            "width": "100%",
                                          }
                                        }
                                      >
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          isTVSelectable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Object {
                                              "opacity": 1,
                                              "zIndex": 1000,
                                            }
                                          }
                                        >
                                          <View
                                            style={
                                              Object {
                                                "display": "flex",
                                                "flexDirection": "row",
                                              }
                                            }
                                          >
                                            <Text
                                              style={
                                                Object {
                                                  "backgroundColor": "#f9f9f9",
                                                  "borderRadius": 4,
                                                  "color": "#333333",
                                                  "fontSize": 12,
                                                  "left": 16,
                                                  "paddingLeft": 3,
                                                  "paddingRight": 3,
                                                  "top": -7,
                                                }
                                              }
                                            >
                                              Company name
                                            </Text>
                                          </View>
                                        </View>
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "borderBottomWidth": 1,
                                              "borderColor": "#9a9a9a",
                                              "borderRadius": 4,
                                              "borderWidth": 1,
                                              "flexDirection": "row",
                                              "paddingLeft": 16,
                                              "paddingRight": 16,
                                              "position": "absolute",
                                              "transform": Array [
                                                Object {
                                                  "translateX": 0,
                                                },
                                              ],
                                            }
                                          }
                                        >
                                          <TextInput
                                            accessibilityLabel="Company name"
                                            allowFontScaling={true}
                                            autoCapitalize="none"
                                            customStyles={
                                              Object {
                                                "customFocusedLabelStyles": Object {
                                                  "color": "#333333",
                                                },
                                                "customHintStyles": Object {
                                                  "color": "#333333",
                                                },
                                                "customLabelStyles": Object {
                                                  "color": "#333333",
                                                },
                                              }
                                            }
                                            hint="As provided in registration email"
                                            name="clientId"
                                            onBlur={[Function]}
                                            onChangeText={[Function]}
                                            onFocus={[Function]}
                                            onSubmitEditing={[Function]}
                                            placeholderTextColor="#3c3c434c"
                                            rejectResponderTermination={true}
                                            returnKeyType="next"
                                            secureTextEntry={false}
                                            style={
                                              Object {
                                                "alignSelf": "center",
                                                "color": "black",
                                                "flex": 1,
                                                "fontSize": 18,
                                                "height": 56,
                                                "minHeight": 40,
                                              }
                                            }
                                            testID="clientId"
                                            underlineColorAndroid="transparent"
                                            value="clientIdFromParam"
                                          />
                                        </View>
                                      </View>
                                      <Text
                                        customHintStyles={
                                          Object {
                                            "color": "#333333",
                                          }
                                        }
                                        focused={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#333333",
                                              "fontSize": 12,
                                              "left": 16,
                                              "marginBottom": 4,
                                              "marginLeft": 4,
                                              "marginRight": 4,
                                              "marginTop": 4,
                                              "position": "absolute",
                                              "top": 60,
                                            },
                                          ]
                                        }
                                      >
                                        As provided in registration email
                                      </Text>
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Array [
                                        Object {},
                                      ]
                                    }
                                  >
                                    <View
                                      height={100}
                                      style={
                                        Array [
                                          Object {
                                            "height": 100,
                                          },
                                        ]
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "paddingHorizontal": 10,
                                            "paddingLeft": 0,
                                            "paddingRight": 0,
                                            "width": "100%",
                                          }
                                        }
                                      >
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          isTVSelectable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Object {
                                              "opacity": 1,
                                              "zIndex": 1000,
                                            }
                                          }
                                        >
                                          <View
                                            style={
                                              Object {
                                                "display": "flex",
                                                "flexDirection": "row",
                                              }
                                            }
                                          >
                                            <Text
                                              style={
                                                Object {
                                                  "backgroundColor": "#f9f9f9",
                                                  "borderRadius": 4,
                                                  "color": "#333333",
                                                  "fontSize": 12,
                                                  "left": 16,
                                                  "paddingLeft": 3,
                                                  "paddingRight": 3,
                                                  "top": -7,
                                                }
                                              }
                                            >
                                              Email address
                                            </Text>
                                          </View>
                                        </View>
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "borderBottomWidth": 1,
                                              "borderColor": "#9a9a9a",
                                              "borderRadius": 4,
                                              "borderWidth": 1,
                                              "flexDirection": "row",
                                              "paddingLeft": 16,
                                              "paddingRight": 16,
                                              "position": "absolute",
                                              "transform": Array [
                                                Object {
                                                  "translateX": 0,
                                                },
                                              ],
                                            }
                                          }
                                        >
                                          <TextInput
                                            accessibilityLabel="Email address"
                                            allowFontScaling={true}
                                            autoCapitalize="none"
                                            customStyles={
                                              Object {
                                                "customFocusedLabelStyles": Object {
                                                  "color": "#333333",
                                                },
                                                "customHintStyles": Object {
                                                  "color": "#333333",
                                                },
                                                "customLabelStyles": Object {
                                                  "color": "#333333",
                                                },
                                              }
                                            }
                                            keyboardType="email-address"
                                            name="username"
                                            onBlur={[Function]}
                                            onChangeText={[Function]}
                                            onFocus={[Function]}
                                            placeholderTextColor="#3c3c434c"
                                            rejectResponderTermination={true}
                                            returnKeyType="next"
                                            secureTextEntry={false}
                                            style={
                                              Object {
                                                "alignSelf": "center",
                                                "color": "black",
                                                "flex": 1,
                                                "fontSize": 18,
                                                "height": 56,
                                                "minHeight": 40,
                                              }
                                            }
                                            testID="username"
                                            underlineColorAndroid="transparent"
                                            value="b@c.com"
                                          />
                                        </View>
                                      </View>
                                      <Text
                                        customHintStyles={
                                          Object {
                                            "color": "#333333",
                                          }
                                        }
                                        focused={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#333333",
                                              "fontSize": 12,
                                              "left": 16,
                                              "marginBottom": 4,
                                              "marginLeft": 4,
                                              "marginRight": 4,
                                              "marginTop": 4,
                                              "position": "absolute",
                                              "top": 60,
                                            },
                                          ]
                                        }
                                      />
                                    </View>
                                  </View>
                                  <View
                                    style={
                                      Array [
                                        Object {},
                                      ]
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "borderRadius": 3,
                                        }
                                      }
                                    >
                                      <View
                                        accessibilityLabel="Reset password"
                                        accessible={true}
                                        focusable={true}
                                        isTVSelectable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Object {
                                            "opacity": 1,
                                          }
                                        }
                                      >
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "backgroundColor": "#DB0011",
                                              "borderColor": "#2089dc",
                                              "borderRadius": 3,
                                              "borderWidth": 0,
                                              "flexDirection": "row",
                                              "height": 48,
                                              "justifyContent": "center",
                                              "padding": 8,
                                            }
                                          }
                                        >
                                          <Text
                                            style={
                                              Object {
                                                "color": "white",
                                                "fontFamily": "Arial",
                                                "fontSize": 16,
                                                "lineHeight": 20,
                                                "paddingBottom": 1,
                                                "paddingTop": 2,
                                                "textAlign": "center",
                                              }
                                            }
                                          >
                                            Reset password
                                          </Text>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
